src/app/app.component.ts

```diff
@@ -93,3 +148,15 @@ export class HeaderComponent<D> implements OnInit {
   }
 }
 
+@Directive({
+  selector: 'jdn-datepicker',
+  providers: [
+    { provide: ACTIVE_CALENDAR, useValue: 'Gregorian' },
```

Comment:
@mmalerba I created an InjectionToken. Do I have to create a directive per calendar, so the token can be set accordingly? Or can the token's value be defined when the directive is used?

```
<jdn-datepicker "set the token's value">
    <input matInput [matDatepicker]="picker" placeholder="Choose a date" [formControl]="form.controls['dateValue']">
    <mat-datepicker-toggle matSuffix [for]="picker"></mat-datepicker-toggle>
    <mat-datepicker #picker [calendarHeaderComponent]="headerComponent"></mat-datepicker>
    </jdn-datepicker>
```

Reply:
Otherwise this is a good solution to initialise the each dat adapter

It would be nice to check the consistency between the injection token's value nd the actually provided calendar date, if any.

Reply:
Instead of making the value a `string` you could probably make it a `Observable<string>` and then give the directive a `[calendarType]` input and emit on the observable whenever it changes. That way your adapter could get the updates

Reply:
> Instead of making the value a `string` you could probably make it a `Observable<string>` and then give the directive a `[calendarType]` input and emit on the observable whenever it changes. That way your adapter could get the updates

I will try that, thanks! This would have to work dynamically. So whenever the datepicker gets initialized with a new date, its calendar should be set correctly in the date adapter.

(sorry for the delayed reaction, but it is only now that I work on editing date values and not just initializing today's date in Gregorian).